proxy/coco/src/project/create.rs

```diff
@@ -144,29 +208,41 @@ impl<Path: AsRef<path::Path>> Repo<Path> {
             },
         }
     }
-
-    /// Get the full path of the `Repo` creation data.
-    fn full_path(&self) -> PathBuf {
-        match self {
-            Self::Existing { path } => path.as_ref().to_path_buf(),
-            Self::New { name, path } => path.as_ref().join(name),
-        }
-    }
 }
 
 /// The data required for creating a new project.
 #[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
-pub struct Create<Path> {
+pub struct Create<R> {
     /// Description of the project we want to create.
     pub description: String,
     /// The default branch name for the project.
     pub default_branch: OneLevel,
-    /// How the repository should be created or opened.
-    pub repo: Repo<Path>,
+    /// The [`Repo`] to create with. It's left as a generic parameter so that we can have
+    /// `Create<Repo>` and `Create<ValidatedRepo>` for that sweet type safety.
+    pub repo: R,
 }
 
-impl<Path: AsRef<path::Path>> Create<Path> {
+impl Create<Repo> {
+    #![allow(clippy::use_self)]
```

Comment:
```suggestion
#[allow(clippy::use_self)]
impl Create<Repo> {
```

Reply:
Why scope it to the whole `impl` rather than the single function?